sdk/azidentity/auth_file_credential.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type authFileAccesstoken struct {
+	ClientId                       string `json:"clientId"`
+	ClientSecret                   string `json:"clientSecret"`
+	SubscriptionId                 string `json:"subscriptionId"`
+	TenantId                       string `json:"tenantId"`
+	ActiveDirectoryEndpointUrl     string `json:"activeDirectoryEndpointUrl"`
+	ResourceManagerEndpointUrl     string `json:"resourceManagerEndpointUrl"`
+	ActiveDirectoryGraphResourceId string `json:"activeDirectoryGraphResourceId"`
+	SqlManagementEndpointUrl       string `json:"sqlManagementEndpointUrl"`
+	GalleryEndpointUrl             string `json:"galleryEndpointUrl"`
+	ManagementEndpointUrl          string `json:"managementEndpointUrl"`
+}
+
+// Enables authentication to Azure Active Directory using configuration information stored Azure SDK Auth File.
+type AuthFileCredential struct {
+	filePath        string
+	credential      azcore.TokenCredential
+	authfileoptions *TokenCredentialOptions
+}
+
+// Creates an instance of the SdkAuthFileCredential class based on information in given SDK Auth file.
+func NewAuthFileCredential(filePath string, options *TokenCredentialOptions) (*AuthFileCredential, error) {
+	if filePath == "" {
+		return nil, fmt.Errorf("The parameter 'filePath' must set a value")
+	}
+
+	return &AuthFileCredential{filePath: filePath, authfileoptions: options}, nil
+}
+
+// Obtains a token from the Azure Active Directory service, using the specified client detailed specified in the SDK Auth file.
+func (c *AuthFileCredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	_, err := c.ensureCredential()
+	if err != nil {
+		return nil, &AuthenticationFailedError{msg: "Error parsing SDK Auth File", inner: err}
+	}
+
+	return c.credential.GetToken(ctx, opts)
+}
+
+// AuthenticationPolicy implements the azcore.Credential interface on AuthFileCredential.
+func (c *AuthFileCredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	return newBearerTokenPolicy(c, options)
+}
+
+// Ensures that credential information is loaded from the SDK Auth file. This method should be called to initialize.
+func (c *AuthFileCredential) ensureCredential() (azcore.TokenCredential, error) {
```

Comment:
I wonder if it's reasonable to pull all of the logic in `ensureCredential` into `NewAuthFileCredential`.  The fact it's a separate method was tied to how we did this in .NET (where doing blocking IO in a constructor felt a little weird) but it feels a little more idiomatic, but interested in what @jhendrixMSFT and other go folks think.

That would also let us front-load more error returning to when the credential is created instead of deferring until the fist time `GetToken` is called.

Reply:
Since exception was thrown in getToken in other languages. We can do in same way here.